controller/webhook/server.go

```diff
@@ -31,28 +34,33 @@ type Handler func(
 // Server describes the https server implementing the webhook
 type Server struct {
 	*http.Server
-	api      *k8s.API
-	handler  Handler
-	recorder record.EventRecorder
+	api       *k8s.API
+	handler   Handler
+	cert      *tls.Certificate
```

Comment:
TIOLI but storing the cert in an atomic.Value might be safer than protecting it with a mutex.  This is very similar to one of the examples given at https://golang.org/pkg/sync/atomic/#Value

Reply:
Ok that's cool in that it avoids us from declaring both `cert` and its mutex, and have instead just an `atomic.Value` we can use. But the approach should be equally safe, no?

Reply:
I mean safe in terms of preventing programmer error.  It's easy to accidentally write code that accesses the field without first acquiring the mutex (which can lead to a data race) or forgetting to release the mutex (which can lead to a deadlock).  It's much harder to use an atomic value wrong.